docs/backjump.md

```diff
@@ -0,0 +1,23 @@
+## Backjump
+
+We learn new clauses when we get conflicts. In this state given
+assignments of variables are incorrect as they create a conflict.
+In order to fix that we need to make `backjump`. This means, we just
+need to reduce current level. 
+
+### Changing level
+
+We need to choose a `level` such that added clause becomes a unit.
```

Comment:
I'm lost in this description. It does not make any sense of what is actually happening in the SAT solver. Can you please check it and reword?